### ISSUE TITLE
Bump kibana from 6.8.X to 7.5.X 

### ISSUE THREAD

**Issue by pburkholder** [cloud-gov/product#1257]
In order to clear ConMon findings related to content-security-policy, upgrade kibana from 6.8.X to 
 7.5.X (or whatever is current). This may also expose better behavior/performance to our customers. 

## Acceptance Criteria

- [ ] Kibana version is > 7.5.1

--- 

## Security considerations

Staying current is a good thing, and this will clear an OWASP finding for `CSP Scanner: Wildcard Directive` (POAM CG 145)

**Comment by pburkholder** [comment]
Here are some of the commits related to content-security-policy: 
https://github.com/elastic/kibana/commit/7a87f03ec7ab4f680c6c45a2c3df4d63c74600b5
https://github.com/elastic/kibana/commit/7094548bcae8105ae612b17dc6ba668c8acdbabc

**Comment by pburkholder** [comment]
Noting that we are dependent on https://github.com/cloudfoundry-community/logsearch-for-cloudfoundry upstream, which is at 6.8.1

**Comment by pburkholder** [comment]
Also, 6.8.1 has a vuln, https://nvd.nist.gov/vuln/detail/CVE-2019-7621, so we should at least patch upstream for that, then deploy.

**Comment by bengerman13** [comment]
patching upstream is complicated, because we don't have access to their blobstore (afaik).

**Comment by pburkholder** [comment]
I may need to move this out of our "Prioritized Backlog" as there are some other equal-priority POA&Ms bumping their way in that may require less effort to address.

**Comment by RonWilliams** [comment]
Will require a scheduled deployment timeline due to breaking changes. Date TBD based on impact discovery.

**Comment by spgreenberg** [comment]
Status:

Changes have been merged into our forks from the upstream into branches:
- https://github.com/cloud-gov/logsearch-boshrelease/pull/56
- https://github.com/cloud-gov/logsearch-for-cloudfoundry/pull/38

In process resolving manifest changes:
- https://github.com/cloud-gov/cg-deploy-logsearch/tree/v211.1.0

I have reviewed 100 of 134 changes in logsearch-boshrelease. I still need to review changes in logsearch-for-cloudfoundry for changes affecting https://github.com/cloud-gov/cg-deploy-logsearch

**Comment by spgreenberg** [comment]
Manifest changes are in a branch: https://github.com/cloud-gov/cg-deploy-logsearch/tree/v211.1.0

I will open a PR once I confirm the bosh releases build successfully.

**Comment by spgreenberg** [comment]
logs in dev is working. Logs-platform in dev is not yet.

**Comment by spgreenberg** [comment]
7.6.1 is now working in dev (logs and logs-platform). I will upgrade staging on tuesday.

**Comment by spgreenberg** [comment]
To Do:

- [ ] Manually upload the elements in the `upload-kibana-objects` job logs-platform . It fails in bosh b/c platform uses opslogin.fr.cloud.gov

- [ ] Create dashboards in staging and test if they are migrated in the upgrade.

**Comment by spgreenberg** [comment]
logs.dev, logs-platform.dev, and logs-platform.staging are updated. However, the logs.dev update experienced the following strange behavior:

- We updated the curator release to address a compilation issue in bosh
- The updated deployment was rolled out successfully (tests passed) to the three environments listed below.
- I merged optimizations made during our last scaling exercise (3 months prior). This included disabling swap on the data nodes. 
- These deployments would then complete in concourse, but fail to start the data nodes correctly. This was due to insufficient memory. This caused the cluster to be non-functional. 
- Forced deletion was required before scaling the node size.

**Comment by spgreenberg** [comment]
To prepare for the production deployment, we should deploy to logs.staging and logs-platform.prod before Thursday. I have preemtively scaled logs.staging data nodes to prevent the issue above: https://github.com/cloud-gov/cg-deploy-logsearch/pull/240. The logs-platform.prod and logs.prod configs should be ready as they are in repo. Because I will be OOO, I will document what I think needs to be done below:

**logs.staging**: 
- This should be ready to go. Unpause the job in concourse.
- When complete, test importing saved kibana objects from Prod. 
  - To export the saved objects from prod, go to https://logs.fr.cloud.gov. Go to `Management` > `Saved Objects` (under the Kibana menu). Click `Export X objects` in the upper right and `Export All`. Save this file somewhere.
  - To import the saved objects into staging, go to https://logs.fr-stage.cloud.gov. Go to `Management` > `Saved Objects` (under the Kibana menu) > 'Import'. Upload the file you exported above. 
  - If it works, you should see dashboards from Prod in staging.
- Remove the `update` block here: https://github.com/cloud-gov/cg-deploy-logsearch/blob/457352291279b86ec82914e169a602004b9c2297/logsearch-staging.yml#L8. We needed 2 canaries and 2 in flight to ensure the new 7.6.1 cluster would hit quorum. Once updated, we don't need this anymore.

**logs-platform.prod**
- This should be ready to go. Unpause the job.
- Because the `Upload Kibana Objects` job does not work in logs platform, we need to export/import the dashboards from staging. 
  - To export the saved objects from staging, go to https://logs-platform.fr-stqage.cloud.gov. Go to `Management` > `Saved Objects` (under the Kibana menu). Click `Export X objects` in the upper right and `Export All`. Save this file somewhere.
  - To import the saved objects into prod, go to https://logs-platform.fr.cloud.gov. Go to `Management` > `Saved Objects` (under the Kibana menu) > 'Import'. Upload the file you exported above. 
  - If it works, you should see dashboards from Prod in staging.
- Remove the `update` block here: https://github.com/cloud-gov/cg-deploy-logsearch/blob/457352291279b86ec82914e169a602004b9c2297/logsearch-platform-production.yml#L26. We needed 2 canaries and 2 in flight to ensure the new 7.6.1 cluster would hit quorum. Once updated, we don't need this anymore.

**logs.prod**
- Grab a last minute export of saved objects.
  - To export the saved objects from prod, go to https://logs.fr.cloud.gov. Go to `Management` > `Saved Objects` (under the Kibana menu). Click `Export X objects` in the upper right and `Export All`. Save this file somewhere.
- Unpause the deployment job
- Import the saved objects into the new cluster.
  - To import the saved objects into prod, go to https://logs.fr.cloud.gov. Go to `Management` > `Saved Objects` (under the Kibana menu) > 'Import'. Upload the file you exported above. 
- Remove the `update` block here:  https://github.com/cloud-gov/cg-deploy-logsearch/blob/457352291279b86ec82914e169a602004b9c2297/logsearch-production.yml#L7. We needed 2 canaries and 2 in flight to ensure the new 7.6.1 cluster would hit quorum. Once updated, we don't need this anymore.

**Comment by ccostino** [comment]
@soutenniza and I worked on this today and got as far as getting the deploy to work and the new version up and running in staging.  This required stopping the VMs running the older version in order to get the deploy to work.

The part we are trying to figure out and understand now is importing the indexes and seeing logs appear in Kibana.  The logs themselves are there and we see a listing of indexes, but they're not flowing into the viewer or dashboard.  We thought this might be an admin permissions issue in staging but that doesn't appear to be the case.

**Comment by ccostino** [comment]
@soutenniza figured out were were just dealing with a permissions issue... D'oh!  We needed to make sure we were admins in staging and had permissions set in the `cloud-gov-operators` org.

**Comment by ccostino** [comment]
@soutenniza and I were able to successful update `logs-platform` in production!  One thing to note:  you should stop all running processes on the `kibana/1` VM with `monit stop all` before starting so the deployment can proceed without getting stuck on the Kibana update itself.

All that's left is the customer-facing logs instance!
